### PR TITLE
Fix Object ID format to be compatible to ECS Atmos

### PR DESCRIPTION
**Pull request by xiaoxin-ren**
1. Fix EsuApiTest build failure in VS2019.
2. Fix Object ID format to be compatible to ECS Atmos.
3. Added support for retention/expiration period header.

### CODE REVIEW

**Review by twincitiesguy — CHANGES_REQUESTED** [review]
`MetadataList` is for user metadata.  To add headers, given the SDK design, we will need to modify the appropriate methods that accept these new headers and add them.  Then re-add the previous method signature to maintain compatibility.

File: EsuApiLib/Rest/EsuRestApi.cs [hunk 1]
```diff
@@ -3885,10 +3885,15 @@ public void DeleteSubtenant(string subtenantId)
                     }
                     listable.Append( formatTag( meta ) );
                 } else {
-                    if( nonListable.Length > 0 ) {
-                        nonListable.Append( ", " );
+                    if (meta.Name == "x-emc-expiration-period" || meta.Name == "x-emc-retention-period") {
```
[the remark this answers is not in the document]
Comment (xiaoxin-ren) [reply EsuApiLib/Rest/EsuRestApi.cs:3888]: Explained as above. .NET version is less OO comparing to Java version. `CustomHeaders ` is intended for 3rd party headers(.e.g auth proxy), and not a good place for native x-emc headers.

[the remark this answers is not in the document]
Comment (twincitiesguy) [reply EsuApiLib/Rest/EsuRestApi.cs:3888]: Based on our zoom discussion, I think we have a good approach - let's add these as properties to `MetadataList`.  That way, we don't need a bunch of new method permutations, and we don't hijack the user-metadata dictionary.

File: EsuApiLib/MetadataList.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright � 2008, EMC Corporation.
+// Copyright ?2008, EMC Corporation.
```
Comment (twincitiesguy) [inline EsuApiLib/MetadataList.cs:1]: Please make sure this is 2022 - the strange character may be due to your editor not using UTF-8 for the file

**Review by twincitiesguy — APPROVED** [review]

**Pull request merged**